### ISSUE TITLE
yerror bug with pyplot()

### ISSUE THREAD

**Issue by ghuba** [JuliaPlots/Plots.jl#1188]
```
julia> using Plots
julia> pyplot()
julia> x = [1, 2, 3]; y = x.^2; e = 0.1 * [1, 1, 1];
julia> plot(x,y)
```
Works

Fails with yerror added:
```
julia> plot(x,y,yerror=e)
Error showing value of type Plots.Plot{Plots.PyPlotBackend}:
ERROR: PyError (ccall(@pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, arg, kw)) <type 'exceptions.TypeError'>
TypeError("'Path' object does not support indexing",)
  File "/Users/.../Library/Python/2.7/lib/python/site-packages/matplotlib/__init__.py", line 1710, in inner
    return func(ax, *args, **kwargs)
  File "/Users/.../Library/Python/2.7/lib/python/site-packages/matplotlib/axes/_axes.py", line 4079, in scatter
    marker_obj = mmarkers.MarkerStyle(marker)
  File "/Users/.../Library/Python/2.7/lib/python/site-packages/matplotlib/markers.py", line 189, in __init__
    self.set_marker(marker)
  File "/Users/.../Library/Python/2.7/lib/python/site-packages/matplotlib/markers.py", line 256, in set_marker
    marker[1] in (0, 1, 2, 3)):
```

The same command works with the plotlyjs() backend.

Any clue?

**Comment by ghuba** [comment]
Just tried on next.juliabox.com and same error

```
PyError (ccall(@pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, arg, kw)) <type 'exceptions.TypeError'>
TypeError("'Path' object does not support indexing",)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/__init__.py", line 1710, in inner
    return func(ax, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_axes.py", line 4079, in scatter
    marker_obj = mmarkers.MarkerStyle(marker)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/markers.py", line 189, in __init__
    self.set_marker(marker)
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/markers.py", line 256, in set_marker
    marker[1] in (0, 1, 2, 3)):
```

Input code:
```
using Plots
pyplot()
n = 32
x = linspace(-1,1,n)
y = x.^2
e = 0.05 * ones(n)
plot(x,y,yerror=e)
```

**Comment by apalugniok** [comment]
I can't replicate the error but I'm pretty sure I'm using python 3.6.

**Comment by mkborregaard** [comment]
I can't reproduce the error either. What's your PyPlot and PyCall versions, and python and matplotlib?

**Comment by ghuba** [comment]
PyCall: 1.15.0
PyPlot: 2.3.2
```
INFO: PyCall is using /usr/bin/python (Python 2.7.10) at /usr/bin/python, libpython = /System/Library/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7
```

```
$ /usr/bin/python
Python 2.7.10 (default, Jul 15 2017, 17:16:57) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import matplotlib
>>> matplotlib.__version__
'2.1.0'
```

OSX 10.13 High Sierra

**Comment by mkborregaard** [comment]
Ah - I wonder if it's High Sierra playing games= My setup is pretty identical otherwise.

**Comment by gdmsl** [comment]
Just got the same error on Archlinux with python-3.6.3, matplotlib-2.1.0, and julia-0.6.1 on a fresh installation of all the packages (deleted my .julia folder and started from scratch).

```julia-repl
julia> using Plots
 
julia> pyplot()
Plots.PyPlotBackend()

julia> plot(rand(10), rand(10), yerror=rand(10))
QApplication: invalid style override passed, ignoring it.
Error showing value of type Plots.Plot{Plots.PyPlotBackend}:
ERROR: PyError (ccall(@pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, arg, kw)) <class 'TypeError'>
TypeError("'Path' object does not support indexing",)
  File "/usr/lib/python3.6/site-packages/matplotlib/__init__.py", line 1710, in inner
    return func(ax, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/matplotlib/axes/_axes.py", line 4079, in scatter
    marker_obj = mmarkers.MarkerStyle(marker)
  File "/usr/lib/python3.6/site-packages/matplotlib/markers.py", line 189, in __init__
    self.set_marker(marker)
  File "/usr/lib/python3.6/site-packages/matplotlib/markers.py", line 256, in set_marker
    marker[1] in (0, 1, 2, 3)):

julia> Pkg.installed("PyPlot")
v"2.3.2"

julia> Pkg.installed("PyCall")
v"1.15.0"

julia> Pkg.installed("Plots")
v"0.13.1"
```

**Comment by kaslusimoes** [comment]
Also happening on my ArchLlinux with packages python-3.6.4-2 and python-matplotlib-2.1.2-1

```julia
julia> versioninfo()
Julia Version 0.6.2
Commit d386e40 (2017-12-13 18:08 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-3930K CPU @ 3.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libm
  LLVM: libLLVM-3.9.1 (ORCJIT, sandybridge)

julia> using Plots; pyplot()
Plots.PyPlotBackend()

julia> plot(rand(10), rand(10), yerror=rand(10))
Error showing value of type Plots.Plot{Plots.PyPlotBackend}:
ERROR: PyError (ccall(@pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, arg, kw)) <class 'TypeError'>
TypeError("'Path' object does not support indexing",)
  File "/usr/lib/python3.6/site-packages/matplotlib/__init__.py", line 1717, in inner
    return func(ax, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/matplotlib/axes/_axes.py", line 4015, in scatter
    marker_obj = mmarkers.MarkerStyle(marker)
  File "/usr/lib/python3.6/site-packages/matplotlib/markers.py", line 189, in __init__
    self.set_marker(marker)
  File "/usr/lib/python3.6/site-packages/matplotlib/markers.py", line 256, in set_marker
    marker[1] in (0, 1, 2, 3)):


julia> Pkg.installed("PyPlot")
v"2.3.2"

julia> Pkg.installed("PyCall")
v"1.15.0"

julia> Pkg.installed("Plots")
v"0.15.0"
```

**Comment by daschw** [comment]
I can reproduce the error using my Archlinux python:
```
$ /usr/bin/python
Python 3.6.4 (default, Jan  5 2018, 02:35:40) 
[GCC 7.2.1 20171224] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import matplotlib
>>> matplotlib.__version__
'2.1.2'
>>> 
```
The code works with Conda.jl python
```julia
ENV["PYTHON"] = ""
Pkg.build("PyPlot")
```

**Comment by daschw** [comment]
Should be fixed in #1411. The issue was the `:hline` markershape passed in the `yerror` recipe.
